headers/h_exception.hpp

```diff
@@ -16,34 +16,48 @@
 
 #include <string>
 #include <iostream>
+#include <exception>
 
 //-----------------------------------------------------------------------
 /*! \brief Exception class.
  *
  *  Sends string (message), line, function and file up the call chain.
  */
-class h_exception /* : public std::exception */ {
-public:
-	std::string extractFilename( const std::string& path )
-	{
-		unsigned long pos = path.find_last_of( '/' );						// *nix
+class h_exception : public std::exception {
+  private:
+	const std::string extractFilename( const std::string& path ) const {
+		unsigned long pos = path.find_last_of( '/' );           // *nix
 		if( pos == path.npos ) pos = path.find_last_of( '\\' );	// win
 		return path.substr( ( pos == path.npos ) ? 0 : pos+1 );
         // degrades gracefully if no separators found
 	}
-	std::string msg, func, file, fullfile;
+	std::string msg, func, file;
 	int linenum;
+
+  public:
+    h_exception() : linenum(0) {}
+    h_exception(const std::string& msg_p, const std::string& func_p,
+                const std::string& file_p, const int linenum_p)
+            : msg(msg_p), func(func_p), file(file_p), linenum(linenum_p) {
+    }
+    virtual ~h_exception() throw() {}
+    inline const std::string get_filename() const {
```

Comment:
Again, doesn't make sense to const qualify a return value that is neither a pointer nor a reference.  Some quick googling suggests that there is some C++ textbook out there that recommends this, but that just goes to show you shouldn't believe everything you read.